lua/legendary/data/function.lua

```diff
@@ -43,4 +87,19 @@ function Function:frecency_id()
   return string.format('<function> %s', self.description)
 end
 
+--- Intended to be used by UI filters
+function Function:modeSwitched()
+  return self._mode_switched
+end
+
+function Function:modes()
+  if self:modeSwitched() then
+    return self.mode_mappings
+  else
+    -- Just use all modes for UI filtering
+    -- it's half-assed because UI filtering is ugly ¯\_(ツ)_/¯
+    return { 'n', 'V', 'v', 'x', 's', 'o', '' }
+  end
```

Comment:
let's keep it consistent with how the `mode_mappings` API works for other items.

```suggestion
  if vim.tbl_islist(self.mode_mappings) then
    return self.mode_mappings
  end

  local modes = {}
  for mode, mapping in pairs(self.mode_mappings) do
    if mapping then
      table.insert(modes, mode)
    end
  end

  if #modes == 0 then
    return { 'n', 'V', 'v', 'x', 's', 'o', '' }
  end

  return modes
```

Reply:
Disagree.

Reply:
Not an intent of this PR; I'm fine if you refactor it further after merging it;

there is a lof of space for improvement in this project